src/main/java/com/squareup/kotlinpoet/Util.kt

```diff
@@ -98,15 +98,15 @@ internal fun stringLiteralWithQuotes(value: String): String {
         i += 2
       } else if (c == '\n') {
         // Add a '|' after newlines. This pipe will be removed by trimMargin().
-        result.append("\n|")
+        result.append("\n" + indent + "|")
       } else {
         result.append(c)
       }
       i++
     }
     // If the last-emitted character wasn't a margin '|', add a blank line. This will get removed
     // by trimMargin().
-    if (!value.endsWith("\n")) result.append("\n")
+    if (!value.endsWith("\n")) result.append("\n" + indent)
```

Comment:
String concatenation shouldn't occur in calls to `append` on a `StringBuilder`. Use multiple calls to `append`